metaflow/monitor.py

```diff
@@ -2,170 +2,158 @@
 
 from contextlib import contextmanager
 
-from .sidecar import SidecarSubProcess
-from .sidecar_messages import Message, MessageTypes
+from metaflow.sidecar import Message, MessageTypes, Sidecar
 
 COUNTER_TYPE = "COUNTER"
 GAUGE_TYPE = "GAUGE"
-MEASURE_TYPE = "MEASURE"
 TIMER_TYPE = "TIMER"
 
 
-class NullMonitor(object):
-    def __init__(self, *args, **kwargs):
-        pass
+class BaseMonitor(object):
+    TYPE = "nullSidecarMonitor"
 
-    def start(self):
-        pass
+    def __init__(self, flow=None, env=None):
+        self._sidecar = Sidecar(self.TYPE)
 
-    @contextmanager
-    def count(self, name):
-        yield
-
-    @contextmanager
-    def measure(self, name):
-        yield
-
-    def gauge(self, gauge):
-        pass
+    def start(self):
+        return self._sidecar.start()
```

Comment:
Wouldn't you want to define the count, measure, gauge methods in this super class?

Reply:
They are, they start at line 28 in this new version of the file.